src-ts/lib/table/Table.tsx

```diff
@@ -0,0 +1,61 @@
+import classNames from 'classnames'
+
+import TableCell from './table-cell/TableCell'
+import { TableColumn } from './table-column.model'
+import styles from './Table.module.scss'
+
+interface TableProps<T> {
+    readonly columns: Array<TableColumn<T>>
+    readonly data: Array<T>
+    readonly onRowClick?: (data: T) => void
+}
+
+const Table: <T extends { [propertyName: string]: any }>(props: TableProps<T>) => JSX.Element
+    = <T extends { [propertyName: string]: any }>(props: TableProps<T>) => {
```

Comment:
Oh, boy. I have a lot to learn about TS and what this line actually says. When it's convenient, I'll have you 'splain it to me.

Reply:
T is generic in typescript

react requires that T always extends an object type for component definitions.

so this just says that T will always be a dictionary with values of type any (aka any json object).  so if you tried to pass an array type as T, it would fail.